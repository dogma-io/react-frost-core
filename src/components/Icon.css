@import '../styles/colors.css';

svg {
  &.frost-icon {
    fill: currentColor;

    &-frost {
      &-add {
        color: var(--frost-color-white);
      }

      &-chevron {
        color: var(--frost-color-grey-5);
      }

      &-close-circle {
        color: var(--frost-color-blue-1);
      }

      &-link {
        color: var(--frost-color-white);
      }

      &-search {
        color: var(--frost-color-grey-4);
      }
    }
  }
}
