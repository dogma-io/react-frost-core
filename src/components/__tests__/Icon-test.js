import Icon, {ICON_ADD} from '../Icon'
import {mount} from 'enzyme'
import React from 'react'

const TESTS = [
  {
    desc: 'when icon set',
    props: {icon: ICON_ADD},
  },
  {
    desc: 'when className and icon set',
    props: {
      className: 'foobar',
      icon: ICON_ADD,
    },
  },
  {
    desc: 'when icon and pack set',
    props: {
      icon: ICON_ADD,
      pack: 'fire',
    },
  },
  {
    desc: 'when className, icon, and pack set',
    props: {
      className: 'foobar',
      icon: ICON_ADD,
      pack: 'fire',
    },
  },
]

describe('Icon', () => {
  TESTS.forEach(({desc, props}) => {
    describe(desc, () => {
      let wrapper

      beforeEach(() => {
        wrapper = mount(<Icon {...props} />)
      })

      it('functions as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
