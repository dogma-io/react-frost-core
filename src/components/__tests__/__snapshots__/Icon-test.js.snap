// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon when className and icon set functions as expected 1`] = `
<Icon
  className="foobar"
  icon="add"
>
  <svg
    className="frost-icon-frost-add foobar"
  >
    <use
      xlinkHref="frost-pack.svg#sprite-add"
    />
  </svg>
</Icon>
`;

exports[`Icon when className, icon, and pack set functions as expected 1`] = `
<Icon
  className="foobar"
  icon="add"
  pack="fire"
>
  <svg
    className="frost-icon-fire-add foobar"
  >
    <use
      xlinkHref="fire-pack.svg#sprite-add"
    />
  </svg>
</Icon>
`;

exports[`Icon when icon and pack set functions as expected 1`] = `
<Icon
  icon="add"
  pack="fire"
>
  <svg
    className="frost-icon-fire-add"
  >
    <use
      xlinkHref="fire-pack.svg#sprite-add"
    />
  </svg>
</Icon>
`;

exports[`Icon when icon set functions as expected 1`] = `
<Icon
  icon="add"
>
  <svg
    className="frost-icon-frost-add"
  >
    <use
      xlinkHref="frost-pack.svg#sprite-add"
    />
  </svg>
</Icon>
`;
